### PR TITLE
Audio modulator support

### PR DESCRIPTION
**Pull request by microsoftsam**
Audio modulator support

Fix issues with rtlcss and autoprefixer. 
Added cleancss to the css pipeline (OSS request approved)
Support for custom download text and download icon

In target: 
Support for custom deploy async method
Support for specifying a custom before compile method
Support for building editor extensions with module loaders and browserify
UI components for audio modulator

### CODE REVIEW

File: webapp/src/cmds.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 /// <reference path="../../built/pxtlib.d.ts"/>
+/// <reference path="../../localtypings/modulator.d.ts"/>
```
Comment (mmoskal) [inline webapp/src/cmds.ts]: Should this file be checked in?

File: cli/cli.ts [hunk 2]
```diff
@@ -1213,7 +1213,15 @@ export function buildTargetAsync(): Promise<void> {
         .then(buildTargetCoreAsync)
         .then(buildSemanticUIAsync)
         .then(() => buildFolderAsync('cmds', true))
-        .then(() => buildFolderAsync('editor', true, 'editor'))
+        .then(() => {
+            if (fs.existsSync(p + "/tsconfig.json")) {
```
Comment (pelikhan) [inline cli/cli.ts]: use ``path.join(`` in future

File: webapp/src/app.tsx [hunk 3]
```diff
@@ -1549,6 +1552,7 @@ ${compileService ? `<p>${lf("{0} version:", "C++ runtime")} <a href="${Util.html
         const trace = run && simOpts.enableTrace;
         const fullscreen = run && !inTutorial && !simOpts.hideFullscreen
         const audio = run && !inTutorial && targetTheme.hasAudio;
+        const useModulator = pxt.appTarget.compile.useModulator;
```
Comment (pelikhan) [inline webapp/src/app.tsx]: compile might be undefined

File: theme/common.less [hunk 4]
```diff
@@ -716,6 +716,12 @@ Avatar
     width: 0;
 }
 
+/* Modulator */
```
Comment (pelikhan) [inline theme/common.less]: why do we need this?

File: webapp/src/app.tsx (hunk 3 above)
Comment (mmoskal) [inline webapp/src/app.tsx]: no, it cannot https://github.com/Microsoft/pxt/blob/master/pxtlib/main.ts#L18

File: theme/common.less (hunk 4 above)
Comment (microsoftsam) [inline theme/common.less]: UI elements for the modulator are defined in pxt. and we should hide them for all targets that don't need them.

File: webapp/src/cmds.ts (hunk 1 above)
Comment (microsoftsam) [inline webapp/src/cmds.ts]: Moved to target.

Comment (microsoftsam) [inline webapp/src/cmds.ts]: removing

**Pull request merged**